legend-engine-core/legend-engine-core-executionPlan-execution/legend-engine-executionPlan-execution/src/main/java/org/finos/legend/engine/plan/execution/nodes/helpers/freemarker/FreeMarkerExecutor.java

```diff
@@ -82,7 +84,12 @@ else if (v instanceof StreamingResult)
     public static String processRecursively(String input, Map<String, ?> variableMap, String templateFunctions)
     {
         String result = process(input, variableMap, templateFunctions);
-        if (!result.equals(input.replace("\\\"", "\"")))
+        String regex = "(\\$\\{)(.+?)(\\})";
```

Comment:
are you sure the RegEx pattern is not too inclusive? also, should probably be compiled once.